zqd/search/endpoint_test.go

```diff
@@ -0,0 +1,59 @@
+package search_test
```

Comment:
I don't see a reason for this to be a different package from the other files in this directory.

Reply:
I was follow the go convention of placing tests in a separate `*_test` package. I like doing this especially when I'm building libs because it allows me to use my package as an end user would, and also so I avoid testing internal functionality.

Reply:
I don't think having a separate *package* ending in `_test` is a go convention. I see that pattern in the zq repo a couple of times, once in the `proc` package, to avoid a dependency, but that's not a common practice I see in well known go systems.

Reply:
I'm seeing it here in the go standard library: https://github.com/golang/go/blob/master/src/sort/search_test.go

Reply:
Its also nice because you don't populate the package namespace with any test-related internal functions.

Reply:
Also mentioned in the [documentation](https://golang.org/cmd/go/#hdr-Testing_flags) for the `go test` command:

> Test files that declare a package with the suffix "_test" will be compiled as a separate package, and then linked and run with the main test binary.

Reply:
I just verified how frequently the `zq` repo uses this pattern, and see it is used in many of the packages here. I don't want to pollute your PR with comments about this, and see this is consistent with the repo as-is. I do think we should change, but I'll move that discussion off of your PR.